addons/mrp/models/mrp_workorder.py

```diff
@@ -169,12 +169,19 @@ def _onchange_finished_lot_id(self):
         """When the user changes the lot being currently produced, suggest
         a quantity to produce consistent with the previous workorders. """
         previous_wo = self.env['mrp.workorder'].search([
-            ('next_work_order_id', '=', self.id)
+            ('next_work_order_id', '=', self.ids[0])
```

Comment:
Why this change ?

Reply:
At onchange self is like mrp.workorder(<NewId origin=4>,) here. And is not working.

Before this, correct qty_producing was not suggested according to selected lot.
In this -
For example if we have MO with 2 WO, then if production is done as-
WO1 - L001 -> 2 Qty produced
            L002 -> 3 Qty produced
Then, in WO2 if user selects L001 then 2 should be suggested qty_producing and 3 if L002 is selected.